docs/developer-docs/latest/developer-resources/plugin-api-reference/server.md

```diff
@@ -35,7 +35,7 @@ This function is called to load the plugin, even before the application is actua
 **Example:**
 
 ```js
-// path ./strapi-server.js
+// path ./src/plugins/my-plugin/strapi-server.js
```

Comment:
Not that I want to be that guy, but shouldn't we document using the same methods that our CLI generates?

EG when you use `yarn strapi generate plugin` the structure it dumps is as such:

`./src/plugins/my-plugin/strapi-server.js`:

```js
'use strict';

module.exports = require('./server');
```

`./src/plugins/my-plugin/server/index.js`:

```js
'use strict';

const register = require('./register');
const bootstrap = require('./bootstrap');
const destroy = require('./destroy');
const config = require('./config');
const contentTypes = require('./content-types');
const controllers = require('./controllers');
const routes = require('./routes');
const middlewares = require('./middlewares');
const policies = require('./policies');
const services = require('./services');

module.exports = {
  register,
  bootstrap,
  destroy,
  config,
  controllers,
  routes,
  services,
  contentTypes,
  policies,
  middlewares,
};
```

Reply:
Good point @derrickmehaffy , @pwizla  we should use this path: `./src/plugins/my-plugin/server/index.js` that wasn't clear in the blog article either, sorry about that.

Reply:
Guys, all of this is a bit confusing as for instance even in our own [examples/getstarted]() code we don't have an `server/index.js`

This shows that basically you can do whatever you want.
The role of the documentation is to describe things as simply as possible, not necessarily document every use case.

The CLI uses some complex structure, because they figured out it would work best that way (lots of files, so they started to organize them into folders)

The migration guide does not recommend using the yarn strapi generate command at any point so this is a bit outside of the scope of this PR.

If at some point we decide to always recommend both having a strapi-server.js file and a server/index.js file, then let's go for it but this would require updating it everywhere in the docs.

Reply:
Sorry @pwizla don't mean to confuse or try to complicate things but I think this guide should make references to file paths based on the folder structure below (to Derrick's point I believe this should represent the same structure as the CLI generator). 

(consistency in the docs as a whole is maybe another conversation worth having outside the scope of this PR) 

I'm available at 6pm if you want to hop on discord. Sorry again for any confusion.

```
/plugin
-- /admin
---- /components
---- /pages
---- // etc...
---- index.js
-- /server
---- /config
---- /controllers
---- /routes
---- bootstrap.js
---- // etc...
---- index.js
-- strapi-server.js // require('./server')
-- strapi-admin.js // require('./admin')
```

Reply:
Thanks again Mark! Let's hop in on Discord at 6PM CET and see if we can have Derrick in the conversation as well :-)